File: app/src/main/java/com/example/bookreader/feature/library/model/Breadcrumb.java
package com.example.bookreader.feature.library.model;

/**
 * Navigation breadcrumb showing the folder hierarchy.
 * Clicking a breadcrumb navigates back to that folder.
 *
 * @param id Folder ID to navigate to, null represents "root" or library home
 * @param label Display text for this breadcrumb (e.g., "Documents", "2024 Books")
 */
public class Breadcrumb {
    private final String id;
    private final String label;

    public Breadcrumb(String id, String label) {
        this.id = id;
        this.label = label;
    }

    public String getId() {
        return id;
    }

    public String getLabel() {
        return label;
    }
}

File: app/src/main/java/com/example/bookreader/library/ui/state/Breadcrumb.kt
package com.example.bookreader.library.ui.state

/**
 * Navigation breadcrumb showing the folder hierarchy.
 * Clicking a breadcrumb navigates back to that folder.
 *
 * @param id Folder ID to navigate to, null represents "root" or library home
 * @param label Display text for this breadcrumb (e.g., "Documents", "2024 Books")
 */
data class Breadcrumb(
    val id: String? = null,
    val label: String,
)
